erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1566,6 +1566,37 @@ func (hph *HexPatriciaHashed) createCellGetter(b []byte, updateKey []byte, row,
 	}
 }
 
+// updateKind is a type of update that is being applied to the trie structure.
+type updateKind uint8
+
+const (
+	// updateKindDelete means after we processed longest common prefix, row ended up empty.
+	updateKindDelete updateKind = 0b0
+
+	// updateKindPropagate is an update operation ended up with a single nibble which is leaf or extension node.
+	// We do not store keys with only one cell as a value in db, instead we copy them upwards to the parent branch.
+	// IF updated prefix was present before, operation produces a delete write for it.
+	updateKindPropagate updateKind = 0b01
+
+	// updateKindBranch is an update operation ended up as a branch of 2+ cells.
+	// That does not necessarily means that branch is NEW, it could be an existing branch that was updated.
```

Comment:
Why not make a distinction when you create a new branch?

Reply:
because it does not matter for folding, in both cases whole row should be fully encoded and stored once again. No use for that information.